src/bindings/CMakeLists.txt

```diff
@@ -1,8 +1,8 @@
 # Copyright (C) 2018-2022 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 #
-
 add_subdirectory(c)
```

Comment:
I propose to have C 2.0 API as the main binding and old C API move to legacy folder.

Reply:
BTW why don't we have one c API library with two interfaces?

Reply:
I agree with you,  C 2.0 API should be main binding, but we don't want to mix 2 version C apis in the same library becauser it may bring confusion to customer.

Reply:
Is it a good solution? Legacy API and 2.0 API coexisting will increase library size, it is easy to mislead  user to use them at the same application, which maybe bring lots of compatible problem or bugs,  and legacy API should trend to be deprecated. So we think separate legacy API and 2.0 API should be better.

Reply:
Could you please explain the current logic and names of libraries?

User uses cmake configs to link libraries with own application. For core libraries it looks like if user uses InferenceEngineConfig, he works with legacy API, but if user uses OpenVINOConfig he starts to use new API.
1 physical library but 2 configs allows to avoid misleading. Moreover we can deprecate legacy API without any impact on new one.

Reply:
Actually, we just use a different name for OV 2.0 C API to distinguish legacy C API and keep legacy library co-exist with OV 2.0 C APIs, which have separated libraries and header files. 
It does make sense to use different cmake configs to distinguish legacy API and OV 2.0 API, and it is the good solution for such scenario of 2 difference version APIs co-existing.  But is there a possible that some users use the 2 cmake config at the same time, which may bring some unnecessary problem/trouble?
Anyhow, I totally agree with you, If you think it is necessary, we can adopt the same policy to co-exist legacy C API and OV 2.0 C APIs.